cpp/server/handler.cc

```diff
@@ -49,16 +49,58 @@ namespace {
 const char kJsonContentType[] = "application/json; charset=ISO-8859-1";
 
 
+// Log an incoming HTTP request in the Apache access log format.
+void LogRequest(evhttp_request* req, int http_status, int resp_body_length) {
+  evhttp_connection* conn = evhttp_request_get_connection(req);
+  char* peer_addr;
+  ev_uint16_t peer_port;
+  evhttp_connection_get_peer(conn, &peer_addr, &peer_port);
+
+  string http_verb;
+  switch (evhttp_request_get_command(req)) {
+    case EVHTTP_REQ_DELETE:
+      http_verb = "DELETE";
+      break;
+    case EVHTTP_REQ_GET:
+      http_verb = "GET";
+      break;
+    case EVHTTP_REQ_HEAD:
+      http_verb = "HEAD";
+      break;
+    case EVHTTP_REQ_POST:
+      http_verb = "POST";
+      break;
+    case EVHTTP_REQ_PUT:
+      http_verb = "PUT";
+      break;
+    default:
+      http_verb = "UNKNOWN";
+      break;
+  }
+
+  const string uri(evhttp_request_get_uri(req));
+
+  // Omitted fields 2, 3, 4 are:
+  //  2: identity of the client determined by identd on the clients machine
+  //  3: userid of the person requesting the document as determined by HTTP authentication
+  //  4: the time that the request was received, since Glog timestamps logged lines anyway.
+  LOG(INFO) << string(peer_addr) << " - - - \"" << http_verb << " " << uri << "\" "
```

Comment:
Can probably do without the pretense of common log format (with the " - - - " bit), if we're doing ```LOG(INFO)```? If this later changes to writing to a common log file, we can add them back.